Add support for Debian 12

Justin-Fernbaugh opened the pull request:
PR for adding Debian 12 OS support to this cookbook.

ramereth commented:
A few issues with this:
- Debian should be excluded from the IUS suite/recipe as this is only for EL platforms
- The gitlfs suite fails due to the fact that the test recipe uses `nobody`. Debian uses `nogroup` for the group and does not have a `nobody` user.

ramereth commented on test/cookbooks/osl-git-test/recipes/osl_gitlfs.rb:
```diff
@@ -19,13 +19,25 @@
 # Attributes allow changing values for different test contexts.
 
+
+pseudoGroup = 'nobody'
```
```suggestion
nogroup = platform_family?('debian') ? 'nogroup' : 'nogroup'
```

ramereth commented on test/cookbooks/osl-git-test/recipes/osl_gitlfs.rb:
```diff
@@ -19,13 +19,25 @@
+    mode '0755'
+  end
+end
+
```
This whole block can be removed by what I did above. I'm confused why you're creating the directory however.

Justin-Fernbaugh replied:
The /tmp/bar directory was getting "dubious ownership" git errors when switching from group 'nobody' to 'nogroup' which cancelled the command execution. This is just a way I found around that error

ramereth commented on test/cookbooks/osl-git-test/recipes/osl_gitlfs.rb:
```diff
@@ -19,13 +19,25 @@
   repository 'https://git.osuosl.org/osuosl/test-lfs.git'
   user 'nobody'
-  group 'nobody'
+  group pseudoGroup
```
```suggestion
  group nogroup
```

ramereth commented on test/integration/osl-gitlfs/inspec/osl_gitlfs_spec.rb:
```diff
@@ -1,3 +1,6 @@
+pseudoGroup = 'nobody'
+pseudoGroup = 'nogroup' if os.family == 'debian'
```
Do the same as I suggested above please.

Justin-Fernbaugh replied:
Pushed changes

Justin-Fernbaugh commented on Berksfile:
```diff
@@ -3,6 +3,7 @@ source 'https://supermarket.chef.io'
 cookbook 'osl-repos', git: 'git@github.com:osuosl-cookbooks/osl-repos'
 cookbook 'osl-resources', git: 'git@github.com:osuosl-cookbooks/osl-resources', branch: 'main'
 cookbook 'osl-selinux', git: 'git@github.com:osuosl-cookbooks/osl-selinux'
+cookbook 'apt', git: 'git@github.com:sous-chefs/apt', branch: 'main'
```
Pull from supermarket instead remove git branch

Justin-Fernbaugh replied:
Pushed

ramereth reviewed (changes requested):
Can you please update `default['osl-git']['ius_git_package']` to be `git226`? It seems `git224` was replaced by 226.

Justin-Fernbaugh commented:
> Can you please update `default['osl-git']['ius_git_package']` to be `git226`? It seems `git224` was replaced by 226.

Updated the package to be git226 and updated the tests as well

ramereth commented on attributes/default.rb:
```diff
@@ -2,4 +2,4 @@
 default['osl-git']['secrets']['credentials'] = []
 
-default['osl-git']['ius_git_package'] = 'git224'
+default['osl-git']['ius_git_package'] = 'git226'
```
I might have typoed this, it needs to be `git236`

Justin-Fernbaugh replied:
Ah okay just pushed the changes to reflect the changes

ramereth commented on test/integration/ius/inspec/ius_spec.rb:
```diff
@@ -2,7 +2,7 @@
 end
 
-describe package('git224') do
+describe package('git236') do
```
You need to fix the test below too please.

Justin-Fernbaugh replied:
Updated the test for git version on git236

ramereth reviewed (approved):
Patch bump

Justin-Fernbaugh commented:
!bump patch ~

osuosl-manatee commented:
Jenkins has merged this PR into `master` and has automatically performed a patch-level version bump to v1.10.2.  Have a nice day!
